mobile/src/main/java/org/openhab/habdroid/background/BackgroundUtils.java

```diff
@@ -0,0 +1,169 @@
+     * Creates notification channels for background tasks.
+     * @param context
+     */
+    public static void createNotificationChannels(Context context) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            NotificationManager notificationManager =
+                    context.getSystemService(NotificationManager.class);
+
+            String name = context.getString(R.string.notification_channel_background);
+            String description =
+                    context.getString(R.string.notification_channel_background_description);
+            int importance = NotificationManager.IMPORTANCE_LOW;
+            NotificationChannel channel =
+                    new NotificationChannel(CHANNEL_ID_BACKGROUND, name, importance);
+            channel.setDescription(description);
+            notificationManager.createNotificationChannel(channel);
+
+            name = context.getString(R.string.notification_channel_background_error);
+            description =
+                    context.getString(R.string.notification_channel_background_error_description);
+            importance = NotificationManager.IMPORTANCE_DEFAULT;
+            channel = new NotificationChannel(CHANNEL_ID_BACKGROUND_ERROR, name, importance);
+            channel.setDescription(description);
+            channel.setLightColor(ContextCompat.getColor(context, R.color.openhab_orange));
+            channel.enableLights(true);
+            notificationManager.createNotificationChannel(channel);
+        }
+    }
+
+    public static Notification makeBackgroundNotification(Context context,
+            @StringRes int msgId, @DrawableRes int iconId, boolean isError,
+            NotificationCompat.Action action) {
+        return makeBackgroundNotification(context, context.getString(msgId), iconId, isError,
+                action);
+    }
+
+    /**
+     * Makes notification for background tasks. Sets notification channel, importance and more
+     * depending on the parameters.
+     *
+     * @param context
+     * @param msg Message to show.
+     * @param iconId Icon to show.
+     * @param isError true if it's an error.
+     * @param action Action to show as a button, e.g. "Retry"
+     * @return
+     */
+    public static Notification makeBackgroundNotification(Context context, String msg,
+            @DrawableRes int iconId, boolean isError, NotificationCompat.Action action) {
+        Intent notificationIntent = new Intent(context, MainActivity.class);
+        notificationIntent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP
+                | Intent.FLAG_ACTIVITY_SINGLE_TOP);
+        PendingIntent intent = PendingIntent.getActivity(context, 0,
+                notificationIntent, 0);
+
+        NotificationCompat.Builder nb = new NotificationCompat.Builder(context,
+                isError ? CHANNEL_ID_BACKGROUND_ERROR : CHANNEL_ID_BACKGROUND)
+                .setLargeIcon(drawableToBitmap(context.getResources().getDrawable(iconId)))
```

Comment:
I wouldn't do this, TBH. Large icon is for things like user images, I haven't seen it being used for icons. The fact you only can pass in bitmaps to it reflects that.

Reply:
Sounds fair. I'll remove it